day3/src/main.rs

```diff
@@ -0,0 +1,59 @@
+use std::{collections::HashSet, fs::read_to_string};
+
+fn main() {
+    let input = read_file("input.txt");
+    println!("{:?}", puzzle1(&input));
+    println!("{:?}", puzzle2(&input));
+}
+
+fn puzzle1(input: &str) -> u32 {
+    input
+        .split("\n")
+        .map(|rucksack| {
+            let (first_half, second_half) = rucksack.split_at(rucksack.len() / 2);
+            let first_half_letters_set: HashSet<char> = HashSet::from_iter(first_half.chars());
```

Comment:
Can consider using `collect`

```suggestion
            let first_half_letters_set = first_half.chars().collect::<HashSet<char>>();
```